### DISCUSSION TITLE
How do I assign a "_unmanaged" pointer from a managed COM reference?

### DISCUSSION THREAD

**Discussion by maaron** [microsoft/CsWin32#757] [Q&A]
For context, I'm trying to create a D3D12_RESOURCE_BARRIER struct, which has a field of type ID3D12Resource_unmanaged*.  How do I set this field if all I have is a ID3D12Resource?

**Comment by AArnott** [comment]
Two options: 
1. Use `fixed` on your `ID3D12Resource` to get a pointer to it. This will _not_ work if the pointer must live longer than your `fixed` block.
2. Use [`GCHandle`](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.gchandle?view=net-7.0) to pin the object for an extended period, and use its `AddrOfPinnedObject` method.
